tools/mkldscript.c

```diff
@@ -73,7 +75,8 @@ static void write_ld_script(FILE *fout)
         for (j = 0; j < seg->includesCount; j++)
         {
             if (!seg->includes[j].dataWithRodata)
-                fprintf(fout, "            %s (.data)\n", seg->includes[j].fpath);
+                fprintf(fout, "            %s (.data)\n"
+                              "        . = ALIGN(0x10);\n", seg->includes[j].fpath);
```

Comment:
We might consider adding a comment on the current necessity of section alignment somewhere, since this file is now festooned in `ALIGN(0x10)`s.

Reply:
Yeah, noting somewhere why the `ALIGN(0x10)`s  where added sound like a good idea. Specially considering when the other rodata sections have an explanation

Reply:
do you guys have suggestions for what and where to put the comment (there are alignment changes all over the file, picking a spot is hard)
you guys are more well versed in the reloc issue than I am

Reply:
Given that it's all over the place, it seems like it would be best at the top of the file rather than duplicating the comment everywhere. Really this whole thing could do with documenting, if not refactoring: at the moment it's pretty hard to follow, especially should another project want to use it.

Reply:
agreed there, we also have commented out stuff everywhere. Figure if we were moving forward with the z64 shared build tools, we can run the formatter and clean things up then

Reply:
I discussed this with @AngheloAlf and we came up with the following stopgap for now:
> ```Note: *SECTION ALIGNMENT* Object files built with a compiler such as GCC can by default use narrower alignment for sections than IDO's 0x10. To properly generate relocations relative to section starts, sections currently need to be aligned to 0x10, because otherwise the linker inserts padding in the wrong place.```

It is perhaps overkill, but hopefully it's clear enough.